hibernate-rx-core/src/main/java/org/hibernate/rx/persister/entity/impl/RxAbstractEntityPersister.java

```diff
@@ -281,9 +280,9 @@ default AbstractEntityPersister delegate() {
 							identifierColumnName,
 							Types.INTEGER
 					);
-			return queryExecutor().update( sql, insert.getParametersAsArray(), factory)
+			return queryExecutor().update( sql, insert.getParametersAsArray(), factory )
 					.thenCompose( v -> queryExecutor().selectLong(selectIdSql, new Object[0], factory) )
-					.thenApply(Optional::get);
+					.thenApply( id -> id );
```

Comment:
Do you need this last `thenApply`?

Reply:
Yup, it does a widening cast.

Reply:
(This wouldn't be necessary if Java had declaration-site variance *grumble*.)